### ISSUE TITLE
Is this code complete？

### ISSUE THREAD

**Issue by Xinchungeorb** [ksatolab-uec/rate-adaptation_for_decentralized-fl#1]
I'm sorry to bother you.Thanks for your code.Is this code complete？

**Comment by ksatolab-uec** [comment]
Yes, this code is complete for the rate adaptation; we generated Fig.11 (average rate performance) based on this code.
Please note that we had applied some minor updates to the original code for the readability that may not affect the rate performance.

However, this code DOES NOT contain the learning part.
Please implement it with your favorite library (e.g., PyTorch) if necessary.
